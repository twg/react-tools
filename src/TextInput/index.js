import React, { Component } from 'react'
import css from './style.css'
import PropTypes from 'prop-types'

export const TextInput = props => {
  const inputClasses = [css.input]
  if (props.error) inputClasses.push(css.error)

  return (
    <div className={css.textinput}>
      <label className={css.inputlabel} htmlFor={props.id}>
        <h5>{props.label}</h5>
      </label>
      <div style={{ ...props.style, position: 'relative' }}>
        <input className={inputClasses.join(' ')} type="text" {...props} />
        {props.error && <div className={css.exclamation}>!</div>}
        {props.error && (
          <div className={css.error}>
            <p>{props.error}</p>
          </div>
        )}
      </div>
    </div>
  )
}

TextInput.propTypes = {
  /**
   * The id provided to the underlying input
   */
  id: PropTypes.string,
  /**
   * The error message displayed
   */
  error: PropTypes.string,
  /**
   * Placeholder text passed to the underlying input
   */
  placeholder: PropTypes.string,
  /**
   * Specifier for the input type
   */
  type: PropTypes.string,
  /**
   * Label to be displayed and associated with the input
   */
  label: PropTypes.string
}

export default TextInput
