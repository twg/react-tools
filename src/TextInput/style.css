.textinput{
    position: relative;
    width: 300px;
}

.input {
    width: 300px;
    height: 45px;
    border: solid 1px grey;
    font-size: 14px;
    box-sizing: border-box;
    text-align: left;
    padding: 10px;
    color: #000;
}

.input::placeholder {
    color: #9b9b9b;
}

.input:active {
    border-color: green;
    border-width: 2px;
}

.input:focus {
    border-color: green;
    border-width: 2px;
    outline: none;
}

.input:disabled{
    color: grey !important;
}

.input.error{
    border-width: 2px;
    border-color: red;
}

.inputlabel{
    display: block;
    padding-bottom: 7px;
}

.inputlabel h5{
    margin-bottom: 0;
}

.exclamation{
    width: 18px;
    height: 18px;
    background: red;
    position: absolute;
    right: 15px;
    top: 15px;
    border-radius: 50%;
    text-align: center;
    color: white;
}

.error{
    margin-top: 3px;
}

.error p{
    color:red;
}