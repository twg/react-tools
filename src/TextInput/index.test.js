/* eslint-env jest */

import React from 'react'
import TextInput from './'
import renderer from 'react-test-renderer'
import cases from 'jest-in-case'

cases(
  'matches snapshots',
  opts => {
    expect(
      renderer
        .create(
          <div style={{ width: '400px' }}>
            <TextInput {...opts.props} />
          </div>
        )
        .toJSON()
    ).toMatchSnapshot()
  },
  [
    { name: 'normal', props: {} },
    { name: 'disabled', props: { disabled: true } },
    { name: 'error', props: { error: 'This is error' } }
  ]
)
