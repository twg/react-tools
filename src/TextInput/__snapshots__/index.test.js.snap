// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshots disabled 1`] = `
<div
  style={
    Object {
      "width": "400px",
    }
  }
>
  <div
    className="textinput"
  >
    <label
      className="inputlabel"
      htmlFor={undefined}
    >
      <h5 />
    </label>
    <div
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <input
        className="input"
        disabled={true}
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`matches snapshots error 1`] = `
<div
  style={
    Object {
      "width": "400px",
    }
  }
>
  <div
    className="textinput"
  >
    <label
      className="inputlabel"
      htmlFor={undefined}
    >
      <h5 />
    </label>
    <div
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <input
        className="input error"
        error="This is error"
        type="text"
      />
      <div
        className="exclamation"
      >
        !
      </div>
      <div
        className="error"
      >
        <p>
          This is error
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`matches snapshots normal 1`] = `
<div
  style={
    Object {
      "width": "400px",
    }
  }
>
  <div
    className="textinput"
  >
    <label
      className="inputlabel"
      htmlFor={undefined}
    >
      <h5 />
    </label>
    <div
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <input
        className="input"
        type="text"
      />
    </div>
  </div>
</div>
`;
